specification/integration_context.md

```diff
@@ -0,0 +1,91 @@
+<!-- markdownlint-configure-file 
+{ 
+  "MD013": { 
+    "tables": false 
+  } 
+}
+-->
+
+# Integration Context
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+In order to create a more complete picture of a distributed environment, we
+propose the use of additional headers to pass extra context details between
+Cisco observability components. Specifically, we are concerned with passing
+information between OpenTelemetry-based Splunk distributions and AppDynamics
+agents, in both directions.
+
+# Headers
+
+The following defines several new headers and describes their intended use.
+All headers MUST be treated as optional -- peer services will not always
+generate them. These headers SHOULD be treated as opaque values of type
+string.
+
+* `cisco-ctx-acct-id` - Contains the ID of the AppDynamics account.
+* `cisco-ctx-app-id` - Contains the ID of the AppDynamics application.
+* `cisco-ctx-tier-id` - Contains the ID of the AppDynamics tier.
+* `cisco-ctx-bt-id` - Contains the ID of the AppDynamics business transaction (BT).
+* `cisco-ctx-env` - Contains
+  the [deployment.environment.name](https://opentelemetry.io/docs/specs/semconv/attributes-registry/deployment/)
+  resource value from an OpenTelemetry based component.
+* `cisco-ctx-service` - Contains the [service.name](https://opentelemetry.io/docs/specs/semconv/resource/#service)
+  resource value from an OpenTelemetry based component.
+
+HTTP headers are capable of being multivalued. As such, implementations
+SHOULD use the _last_ value when the above headers contain multiple values.
+
+## Splunk OpenTelemetry distributions
+
+This section applies to Splunk distributions of OpenTelemetry instrumentation
+components. In brief, Splunk OTel distributions will need to understand when
+spans are being created as part of an AppDynamics "Business Transaction" (BT),
+and will add span attributes to clarify the AppDynamics invocation context.
+
+### Configuration
+
+Splunk OTel distributions MUST provide a way for users to opt-into the
+consumption and propagation of additional Cisco (bespoke) integration context.
+
+The configuration SHOULD be named `cisco.ctx.enabled` or an idiomatic
+equivalent for each language. If using environment variables, the
+configuration SHOULD be named `CISCO_CTX_ENABLED`. The default value
+MUST be `false` or equivalent language-specific non-truthy value.
```

Comment:
Are we sure that it should be disabled by default? Would it not be inconvenient for users to enable it?
Do we even need such configuration?
If a AppD integration/communication would not be present than it would be an "no-op". Requiring to enable may be inconvenient.

Reply:
I think that it might be needed for outgoing requests.

Reply:
I guess the integration somehow still needs to find out which "outgoing requests" are targeting AppD. Otherwise, the headers would leak in all outgoing requests.

Reply:
It's fine that, when enabled, the headers are in outgoing requests. That's intentional. 

What the default of `false` does is prevent non-AppD users from getting these unnecessary headers. Even some AppD users probably don't want it, so the default state should maintain the status quo. It takes extra config/effort to turn on the special/new feature. Yes.

Reply:
Resolving per this conversation which I see as a duplicate: https://github.com/signalfx/gdi-specification/pull/335#discussion_r1965156593